src/bentoml/_internal/cloud/deployment.py

```diff
@@ -1,73 +1,222 @@
+        res = yatai_rest_client.get_deployment(
+            cluster_name,
+            create_deployment_schema.kube_namespace,
+            create_deployment_schema.name,
+        )
+        if res is not None:
+            raise BentoMLException("Create deployment: Deployment already exists")
+        res = yatai_rest_client.create_deployment(
+            cluster_name, schema_to_json(create_deployment_schema)
+        )
+        if res is None:
+            raise BentoMLException("Create deployment request failed")
+        print(f"{create_deployment_schema.name} is created.")
+        return res
+
+    def create_deployment(
+        self,
+        deployment_name: str,
+        bento_repository: Tag | str,
+        description: str = None,
+        cluster_name: str = default_context_name,
+        kube_namespace: str = default_kube_namespace,
+        cpu: str | None = None,
+        gpu: str | None = None,
+        memory: str | None = None,
+        hpa_conf: DeploymentTargetHPAConf | None = None,
+        runners_config: dict[str, DeploymentTargetRunnerConfig] | None = None,
+        api_server_config: DeploymentTargetConfig | None = None,
+        mode: DeploymentMode = DeploymentMode.Function,
+        type: DeploymentTargetType = DeploymentTargetType.STABLE,
+        context: str | None = None,
+        do_not_deploy: bool = False,
+        labels: t.List[dict[str, str]] | None = None,
+        canary_rules: t.List[DeploymentTargetCanaryRule] | None = None,
+    ) -> DeploymentSchema:
+        if isinstance(bento_repository, str):
+            bento_repository = bentoml_cattr.structure(bento_repository, Tag)
+        print(bento_repository)
+        dct = {
+            "name": deployment_name,
+            "kube_namespace": kube_namespace,
+            "mode": mode,
+            "labels": labels,
+            "description": description,
+            "do_not_deploy": do_not_deploy,
+        }
+        if api_server_config is None:
+            dct["targets"] = [
+                {
+                    "type": type,
+                    "bento_repository": bento_repository.name,
+                    "bento": bento_repository.version,
+                    "canary_rules": canary_rules,
+                    "config": {
+                        "resources": {},
+                        "runners": runners_config,
+                    },
+                }
+            ]
+
+        else:
+            api_server_config.runners = runners_config
+            dct["targets"] = [
+                {
+                    "type": type,
+                    "bento_repository": bento_repository.name,
+                    "bento": bento_repository.version,
+                    "canary_rules": canary_rules,
+                    "config": bentoml_cattr.unstructure_attrs_asdict(api_server_config),
+                }
+            ]
+        if cpu or gpu or memory:
+            resource_spec = {"requests": {"cpu": cpu, "gpu": gpu, "memory": memory}}
```

Comment:
This line is not correct right? It should be provisional based on the field that is not None and parsed into resource_spec